src/org/exoplatform/ui/social/SelectedImageActivity.java

```diff
@@ -42,255 +40,226 @@
+    if (view.equals(removeButton)) {
+      if (modeId == EDIT_MODE) {
+        ComposeMessageActivity.removeImageFromMessage();
+      }
+      if (modeId == SELECT_MODE) {
+        backToGallery();
+      }
+    }
+    finish();
 
-        }
-
-        if (view.equals(removeButton)) {
-            if (modeId == EDIT_MODE) {
-                ComposeMessageActivity.removeImageFromMessage();
-            }
-            if (modeId == SELECT_MODE) {
-                backToGallery();
-            }
-        }
-        finish();
+  }
 
+  private void backToGallery() {
+    Intent intent = new Intent(Intent.ACTION_PICK);
+    intent.setType(ExoConstants.PHOTO_ALBUM_IMAGE_TYPE);
+    if (DocumentActivity._documentActivityInstance != null) {
+      DocumentActivity._documentActivityInstance.startActivityForResult(intent, ExoConstants.REQUEST_ADD_PHOTO);
+    } else if (ComposeMessageActivity.composeMessageActivity != null) {
+      ComposeMessageActivity.composeMessageActivity.startActivityForResult(intent, ExoConstants.REQUEST_ADD_PHOTO);
     }
+  }
 
-    private void backToGallery() {
-        Intent intent = new Intent(Intent.ACTION_PICK);
-        intent.setType(ExoConstants.PHOTO_ALBUM_IMAGE_TYPE);
-        if (DocumentActivity._documentActivityInstance != null) {
-            DocumentActivity._documentActivityInstance.startActivityForResult(intent,
-                                                                              ExoConstants.REQUEST_ADD_PHOTO);
-        } else if (ComposeMessageActivity.composeMessageActivity != null) {
-            ComposeMessageActivity.composeMessageActivity.startActivityForResult(intent,
-                                                                                 ExoConstants.REQUEST_ADD_PHOTO);
-        }
-    }
+  private void onChangeLanguage() {
+    Resources resource = getResources();
+    okText = resource.getString(R.string.OK);
+    cancelText = resource.getString(R.string.Cancel);
+    removeText = resource.getString(R.string.Remove);
+    loadingData = resource.getString(R.string.LoadingImage);
+    title = resource.getString(R.string.SelectedPhoto);
 
-    private void onChangeLanguage() {
-        Resources resource = getResources();
-        okText = resource.getString(R.string.OK);
-        cancelText = resource.getString(R.string.Cancel);
-        removeText = resource.getString(R.string.Remove);
-        loadingData = resource.getString(R.string.LoadingImage);
-        title = resource.getString(R.string.SelectedPhoto);
+  }
 
-    }
+  private class DisplayImageTask extends AsyncTask<Integer, Void, Bitmap> {
+    private SeclectImageWaitingDialog _progressDialog;
 
-    private class DisplayImageTask extends AsyncTask<Integer, Void, Bitmap> {
-        private SeclectImageWaitingDialog _progressDialog;
+    @Override
+    public void onPreExecute() {
+      _progressDialog = new SeclectImageWaitingDialog(SelectedImageActivity.this, null, loadingData);
+      _progressDialog.show();
```

Comment:
_progressDialog show() here and dismiss() in onPostExecute but leak the case when Task is canceled.